server/controllers/project.controller.js

```diff
@@ -264,9 +236,11 @@ async function buildZip(project, req, res) {
   }
 }
 
-export function downloadProjectAsZip(req, res) {
-  Project.findById(req.params.project_id, (err, project) => {
-    // save project to some path
-    buildZip(project, req, res);
-  });
+export async function downloadProjectAsZip(req, res) {
```

Comment:
`downloadProjectAsZip`, if the project with the given ID doesn't exist, the function sends a response with a 404 status code but doesn't return or terminate the function. Consider using `return` or `else` to prevent further execution if the project doesn't exist.

Reply:
You are right.  Good catch!